backend/stakepoold/rpcclient.go

```diff
@@ -111,66 +109,68 @@ func connectWalletRPC(cfg *config) (*dcrrpcclient.Client, semver, error) {
 }
 
 func walletFetchUserTickets(ctx *appContext) (map[chainhash.Hash]string, string) {
-	// This is suboptimal to copy and needs fixing.
-	users := make(map[string]userdata.UserVotingConfig)
-	ctx.RLock()
-	for k, v := range ctx.userVotingConfig {
-		users[k] = v
-	}
-	ctx.RUnlock()
+	var (
+		ticketcount, usercount int
+	)
 
 	userTickets := make(map[chainhash.Hash]string)
 
+	log.Infof("Running GetTickets, this may take awhile...")
+	tickets, err := ctx.walletConnection.GetTickets(false)
+	if err != nil {
+		log.Warnf("GetTickets failed: %v", err)
+		ticketNoun := pickNoun(ticketcount, "ticket", "tickets")
+		userNoun := pickNoun(usercount, "user", "users")
+		return userTickets, fmt.Sprintf("loaded %v %v for %v %v",
+			ticketcount, ticketNoun, usercount, userNoun)
+	}
+
 	type promise struct {
-		dcrrpcclient.FutureStakePoolUserInfoResult
-		msa string
+		dcrrpcclient.FutureGetTransactionResult
 	}
-	promises := make([]promise, 0, len(users))
-	for msa, v := range users {
-		addr, err := dcrutil.DecodeAddress(msa, ctx.params)
-		if err != nil {
-			log.Infof("Could not decode multisig address %v for %v: %v",
-				msa, v.Userid, err)
-			continue
-		}
+	promises := make([]promise, 0, len(tickets))
 
-		promises = append(promises, promise{
-			ctx.walletConnection.StakePoolUserInfoAsync(addr), msa})
+	for _, ticket := range tickets {
+		// lookup ownership of each ticket
+		promises = append(promises, promise{ctx.walletConnection.GetTransactionAsync(ticket)})
 	}
 
-	var (
-		ticketcount, usercount int
-	)
+	ctx.RLock()
 
 	for _, p := range promises {
-		spui, err := p.Receive()
+		gt, err := p.Receive()
 		if err != nil {
-			log.Errorf("unable to fetch tickets for user %v multisigaddr %v: %v",
-				users[p.msa].Userid, p.msa, err)
+			// All tickets should exist and be able to be looked up
+			log.Warnf("GetTransaction error: %v", err)
 			continue
 		}
-
-		if spui == nil || len(spui.Tickets) <= 0 {
-			continue
-		}
-
-		for _, ticket := range spui.Tickets {
-			if ticket.Status != "live" {
+		for i := range gt.Details {
+			_, ok := ctx.userVotingConfig[gt.Details[i].Address]
+			if !ok {
+				log.Warnf("Could not map ticket %v to a user, user %v doesn't exist", gt.TxID, gt.Details[i].Address)
 				continue
 			}
 
-			hash, err := chainhash.NewHashFromStr(ticket.Ticket)
+			hash, err := chainhash.NewHashFromStr(gt.TxID)
 			if err != nil {
 				log.Infof("invalid ticket %v", err)
 				continue
 			}
-
-			userTickets[*hash] = users[p.msa].MultiSigAddress
-			ticketcount++
+			userTickets[*hash] = ctx.userVotingConfig[gt.Details[i].Address].MultiSigAddress
```

Comment:
I have to digest this a bit more, but I'm also thinking along the lines of not accidentally voting on malformed tickets (e.g. attempted cheaters) in addition to being sure to vote on the ones that were added manually.